lib/ops/opsservice/configure.go

```diff
@@ -1430,6 +1437,20 @@ func (s *site) addClusterConfig(config clusterconfig.Interface, overrideArgs map
 	return args
 }
 
+// serfKeygen generates an encryption key for use in serf agent
+// https://github.com/hashicorp/serf/blob/a2bba5676d6e37953715ea10e583843793a0c507/cmd/serf/command/keygen.go#L20
```

Comment:
Serf is under the Mozilla Public License 2.0, which is a copyleft-ish licence.  Based on the FAQ:

https://www.mozilla.org/en-US/MPL/2.0/FAQ/#mpl-bsd-and-apache

> May I combine MPL-licensed code and BSD-licensed code in the same executable program? What about Apache?
>
>Yes to both. Mozilla currently does this with BSD-licensed code. For example, libvpx, which is used in Firefox to decode WebM video, is under a BSD license.

> Q9: I want to distribute (outside my organization) MPL-licensed source code that I have modified. What do I have to do?

> To see the complete set of requirements, read the license. However, generally:

> * You must inform the recipients that the source code is made available to them under the terms of the MPL (Section 3.1), including any Modifications (as defined in Section 1.10) that you have created.
> * You must make the grants described in Section 2 of the license.
> * You must respect the restrictions on removing or altering notices in the source code (Section 3.4).

I think we're ok here if we add a note that portions of the codebase are MPL.  That seems like a bunch of work for basically no return though.

Instead, I might suggest that we ask someone else (who hasn't looked at the serf code) to give us a quick keygen function.  E.g. give them the signature`generate32ByteEncryptionKey() (string,error)` and then swap in that code without a link to the serf source.  Unfortunately, I'm now tainted since I reviewed this as is :).

Reply:
Ah thanks for pointing this out. Not something I usually pay much attention to.